src/main/java/org/tom_v_squad/soiwenttoaconcert/controllers/ProfileController.java

```diff
@@ -1,74 +1,96 @@
+        if (userId == null) {
+            return null;
+        }
+        Optional<User> user = userRepository.findById(userId);
+        if (user.isEmpty()) {
+            return null;
+        }
+        return user.get();
     }
-}
 
+    public void setUserRepository(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
 
-//    @Autowired
-//    private UserRepository userRepository;
-//
-//    private static final String userSessionKey = "user";
-//    private User updateProfile;
-//
-//    public User getUserFromSession(HttpSession session) {
-//        Integer userId = (Integer) session.getAttribute(userSessionKey);
-//        if (userId == null) {
-//            return null;
-//        }
-//        Optional<User> user = userRepository.findById(userId);
-//        if (user.isEmpty()) {
-//            return null;
-//        }
-//        return user.get();
-//    }
-//
-//    public void setUserRepository(UserRepository userRepository) {
-//        this.userRepository = userRepository;
-//    }
-//
-//    @GetMapping("")
-//    public String displayMyProfile(HttpServletRequest request, Model model) {
-//        User user = getUserFromSession(request.getSession());
-//        String username = user.getUsername();
-////        String location = profile.getlocation();
-////        String userBio = profile.getBiography();
-//        model.addAttribute("greeting", username);
+
+    // SHOWING THE PROFILE TO THE FRONT END
+
+    @GetMapping("")
+    public String displayMyProfile(HttpServletRequest request, Model model) {
+//        this.biography = biography;
+//        this.location = location;
+        User user = getUserFromSession(request.getSession());
+        String username = user.getUsername();
+//        String bio = user.getUserBiography();
+//        String location = user.getUserLocation();
+        model.addAttribute("greeting", username);
+        model.addAttribute("biography", user.getUserBiography());
+        model.addAttribute("location", user.getUserLocation());
+        return "profile/index";
 //        return "profile/index";
-//    }
-//    //
+    }
+
+    @GetMapping("update")
+    public String displayViewArtist(Model model, @PathVariable int userId) {
+        Optional<User> result = userRepository.findById(userId);
+        if (result.isPresent()) {
+            User profile = result.get();
+            model.addAttribute("profile", profile);
+            return "";
+        } else {
+            return "redirect:../";
+        }
+
+
+        @PostMapping("update")
+        public String processCreateProfile(@ModelAttribute User updateUser,
+                Errors errors, Model model){
+            if (errors.hasErrors()) {
+                model.addAttribute("title", "Update Profile");
+                return "profile/update";
+            }
+
+            User updateUser = null;
+            userRepository.save(updateUser);
+            return "redirect: ";
+        }
+    }
```

Comment:
You've got a method inside a method here.  This is why indentation matters - it makes it a lot easier to keep track of what's supposed to be happening.  I also recommend installing a Rainbow Brackets / Colorizer plug in that adds color to matching pairs of parens, curly braces, square brackets, etc to make closing them easier.